### PR TITLE
fix(wordpress-source-plugin): create nodes properly for images who exceed maxFileSizeBytes

### LINKED ISSUE #30755: bug(gatsby-source-wordpress): maxFileSizeBytes option silently excludes entire MediaItem node instead of just the File node
## Description

The featuredImage node is not working for pages & custom post types. It only works for posts.

Here's my query:

```
query PAGE_QUERY {
  allWpPage {
    nodes {
      id
      featuredImage {
        node {
          altText
          sourceUrl
        }
      }
    }
  }
}
```

And here's the result:

```
{
    "id": "cG9zdDo0NTI=",
    "featuredImage": null // this page does NOT have a featured image
},
{
     "id": "cG9zdDoyOTQ3NzA=",
     "featuredImage": {
         "node": null // this page has a featured image
      }
}
```

The second item in the list has a featured image and notice that it's returning a "node" which is null, whereas the top page does not have a featured image and is just returning null

### Steps to reproduce

1. Create a page with a featured image (or any other custom post type) in WP
2. Run the PAGE_QUERY above in Gatsby's in-browser IDE

### Expected result

For pages with a featured image, the `node` object should not be null. It should return `altText` & `sourceUrl` or whatever properties are specified in the query.

### Actual result

The featuredImage.node object is null.

### Environment

It's worth noting I am using Polylang & ACF plugins in WordPress. However, the WPGraphQL IDE works and returns the featuredImage. So it must be an issue with gatsby-source-wordpress.

I am using `gatsby-source-wordpress@5.3.0-next.2`

```
System:
    OS: macOS Mojave 10.14
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.4.0 - ~/.nvm/versions/node/v14.4.0/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 7.7.6 - ~/.nvm/versions/node/v14.4.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.114
    Firefox: 84.0.2
    Safari: 12.0
```

### PR DESCRIPTION
**Pull request by veryspry**
## Description

The WordPress source plugin was returning incorrectly formatted node data for page/post `featuredImages` whose size exceeded the `maxFileSizeBytes` config option. The result of this was that the `featuredImage.node` field would be set to `null` and not return other relevant info about the featured image.

This means that a query for `allWpPost` or `allWpPage` will no longer return this result:
```js
"allWpPost": {
      "nodes": [
        {
          "id": "cG9zdDo1",
          "featuredImage": {
            // this isn't great because we don't know important info like at what url this resource can be found at
            "node": null
          }
        },
      ]
    }
```
```js
"allWpPost": {
      "nodes": [
        { at
          "id": "cG9zdDo1",
          "featuredImage": {
            // now important node information is returned
            "node": {
              "id": "cG9zdDo2",
              "sourceUrl": "http://wordpresssite.com/wp-content/uploads/2021/05/picture-scaled.jpg",
              "localFile": null
            }
          }
        },
      ]
    }
```

## Related Issues

Fixes #30755

### CODE REVIEW

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js [hunk 1]
```diff
@@ -230,13 +247,25 @@ export const createRemoteMediaItemNode = async ({
     maxFileSizeBytes,
   } = pluginOptions.type?.MediaItem?.localFile
 
+  const maybePrintWarning = maybePrintWarningFactory(reporter)
+
   // if this file is larger than maxFileSizeBytes, don't fetch the remote file
   if (fileSize > maxFileSizeBytes) {
+    maybePrintWarning({
+      message: `At least one remote media item not fetched because its size exceeded the maxFileSizeBytes config option. Other media items items may have been skipped for this reason as well but this warning is only logged once to avoid terminal spam.`,
```
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:255]: Can we allow this to log a few times and add the file name and/or db ID to the warning to help with debugging?

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:255]: Oo the word "items" is printed twice as well :)

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js [hunk 2]
```diff
@@ -230,13 +247,25 @@ export const createRemoteMediaItemNode = async ({
...
+      warningName: `maxFileSizeBytesExceeded`,
+    })
+
     return null
   }
 
   // if this type of file is excluded, don't fetch the remote file
   if (excludeByMimeTypes.includes(mimeType)) {
+    maybePrintWarning({
+      message: `At least one remote media item node not created because its mimetype was specified in the "exludeByMimeTypes" config option. Other media items items may have been skipped for this reason as well but this warning is only logged once to avoid terminal spam.`,
```
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:265]: Same as above here ☝️

**Review by TylerBarnes — CHANGES_REQUESTED** [review]
Awesome work! 🙌 just a couple tiny changes then LGTM!

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js (hunk 1 above)
Comment (veryspry) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:255]: Totally! Does three times sound like the correct amount?

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:255]: I think we could do 10 without being too disruptive. Maybe try logging a bunch out to see what it looks like

File: integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js [hunk 3]
```diff
@@ -308,4 +315,69 @@ describe(`data resolution`, () => {
...
+      }
+    })
+
+    expect(nodes.length).toEqual(1)
+    nodes.forEach(node => {
+      expect(node.localFile).toEqual(null)
+    })
+  })
+
+  it(`Does not dowload files whose mime types are excluded`, async () => {
```
Comment (TylerBarnes) [inline integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js:354]:
```suggestion
  it(`Does not download files whose mime types are excluded`, async () => {
```

File: integration-tests/gatsby-source-wordpress/test-fns/test-utils/get-gatsby-process.js [hunk 4]
```diff
@@ -34,3 +34,8 @@ exports.spawnGatsbyProcess = (command = `develop`, env = {}) =>
       },
     }
   )
+
+  process.on('SIGINT', proc.kill)
```
Comment (TylerBarnes) [inline integration-tests/gatsby-source-wordpress/test-fns/test-utils/get-gatsby-process.js:38]: 🎉

File: integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js [hunk 5]
```diff
@@ -308,4 +315,69 @@ describe(`data resolution`, () => {
...
+  it(`Does not download files whose size exceed the maxFileSizeBytes option`, async () => {
+    const wpPluginOpts = getPluginConfig()
+    const { maxFileSizeBytes } = wpPluginOpts.options.type.MediaItem.localFile
+    /**
+     * Ensure that the fileSize "gt" filter value matches the maxFileSizeBytes value in gatsby-config
+     */
+    const { data: { allWpMediaItem: { nodes }}} = await fetchGraphql({
+      url,
+      query: /* GraphQL */`
+        query tooLargeFiles($maxFileSizeBytes: Int!, $includedMimeTypes: [String]!) {
```
Comment (TylerBarnes) [inline integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js:328]: This is awesome! What do you think about also doing the opposite by querying using `lt` and asserting that all nodes below the maxFileSizeBytes option have a localFile node

File: packages/gatsby-source-wordpress/src/steps/log-post-build-warnings.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { Step } from "./../utils/run-steps"
+import store from "~/store"
+import { formatLogMessage } from "~/utils/format-log-message"
+
+export const logPostBuildWarnings: Step = async (): Promise<void> => {
```
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/log-post-build-warnings.ts:5]: Nice, this is a great idea!

File: packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js [hunk 7]
```diff
@@ -1,3 +1,4 @@
+import os from "os"
```
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/create-nodes/create-local-file-node.js:1]: This must've been an accidental auto-import in vscode!

File: packages/gatsby-source-wordpress/src/steps/log-post-build-warnings.ts (hunk 6 above)
Comment (veryspry) [reply packages/gatsby-source-wordpress/src/steps/log-post-build-warnings.ts:5]: :raised_hands:

File: integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js (hunk 5 above)
Comment (veryspry) [reply integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js:328]: Noice! I just added a test for this as well as asserting that files who have included mime types also have a local file node.

File: integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js (hunk 3 above)
Comment (veryspry) [reply integration-tests/gatsby-source-wordpress/test-fns/data-resolution.js:354]: Fixed!

**Comment by TylerBarnes** [comment]
@veryspry looks like the snapshots need to be updated. For these tests I usually just delete the snapshots and run the tests again since the `test` npm script is a bit awkward to run with `--updateSnapshot`

**Comment by veryspry** [comment]
> @veryspry looks like the snapshots need to be updated. For these tests I usually just delete the snapshots and run the tests again since the `test` npm script is a bit awkward to run with `--updateSnapshot`

Weird, I thought I already did that, but I must have changed the site structure to accommodate one of the most recent assertions :thinking: 

Anyways, snapshots have been updated!

**Review by TylerBarnes — APPROVED** [review]
Looks great :D thanks again for this!

**Pull request merged**